From: Matthieu Baerts <matttbe@gmail.com>
Subject: Fixed a compilation error with GLib 2.33
Bug: https://code.launchpad.net/~matttbe/indicator-applet/glib_error/+merge/114763
Forwarded: yes

Index: indicator-applet_ubuntu-desktop/src/tomboykeybinder.h
===================================================================
--- indicator-applet_ubuntu-desktop.orig/src/tomboykeybinder.h	2012-07-13 00:21:42.782048000 +0200
+++ indicator-applet_ubuntu-desktop/src/tomboykeybinder.h	2012-07-13 02:02:01.221440930 +0200
@@ -18,7 +18,7 @@
 #ifndef __TOMBOY_KEY_BINDER_H__
 #define __TOMBOY_KEY_BINDER_H__
 
-#include <glib/gtypes.h>
+#include <glib.h>
 
 G_BEGIN_DECLS
 
